app/src/main/java/org/piwigo/bg/UploadService.java

```diff
@@ -187,33 +258,49 @@ private String getPhotoName(String imageName) {
         return (requestBodies);
     }
 
-    private void callUploadResponse(ImageUploadQueue<UploadAction> imageUploadQueue, RestService restService, ArrayList<RequestBody> requestBodies,
-                                    int catId, MultipartBody.Part filePart) {
-        Call<ImageUploadResponse> call = restService.uploadImage(requestBodies.get(0), catId, requestBodies.get(1), requestBodies.get(2), filePart);
-        RefreshRequestEvent refreshEvent = new RefreshRequestEvent(catId);
+    private void callChunkedUploadResponse(ImageUploadQueue<UploadAction> imageUploadQueue, RestService restService, UploadPromise promise) {
+
+        RefreshRequestEvent refreshEvent = new RefreshRequestEvent(promise.getCatId());
+
+        //We have uploaded every chunk, let's jump to the next image
+        if (promise.getCurrentChunk() > promise.getTotalChunks() - 1) {
```

Comment:
isn't this why they invented >= ?

Reply:
This seemed legit to me and more understandable as currentChunk starts at 0 and totalChunks at 1 (I reproduced the same behavior than the PHP script), but we could go for '>=' for sure

Reply:
doesn't matter too much. Your arguments are clearly valid.